consensus/core/src/config/mod.rs

```diff
@@ -68,6 +68,9 @@ pub struct Config {
 
     /// A scale factor to apply to memory allocation bounds
     pub ram_scale: f64,
+
+    /// The number of days to keep data for
+    pub retention_period_days: f64,
```

Comment:
my personal preference is that we make this an `Option` (as well as the cmd arg too) and explicitly handle the `None` case in pruning processor by directly setting the retention root as the pruning point. This makes it easier for me to verify that the changes do not affect the default way of using a node and thus reduces risks to zero.